nixos/modules/services/mail/public-inbox.nix

```diff
@@ -0,0 +1,431 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.public-inbox;
+
+  inboxesDir = "/var/lib/public-inbox/inboxes";
+  inboxPath = name: "${inboxesDir}/${name}";
+  gitPath = name: "${inboxPath name}/all.git";
+
+  inboxes = mapAttrs (name: inbox:
+    (recursiveUpdate {
+      inherit (inbox) address url newsgroup watch;
+      mainrepo = inboxPath name;
+      watchheader = inbox.watchHeader;
+    } inbox.config))
+    cfg.inboxes;
+
+  concat = concatMap id;
+
+  configToList = attrs:
+    concat (mapAttrsToList (name': value':
+      if isAttrs value' then
+        map ({ name, value }: nameValuePair "${name'}.${name}" value)
+          (configToList value')
+      else if isList value' then map (nameValuePair name') value'
+      else if value' == null then []
+      else [ (nameValuePair name' value') ]) attrs);
+
+  configFull = recursiveUpdate {
+    publicinbox = inboxes // {
+      nntpserver = cfg.nntpServer;
+      wwwlisting = cfg.wwwListing;
+    };
+    publicinboxmda.spamcheck = cfg.mda.spamCheck;
+    publicinboxwatch.spamcheck = cfg.watch.spamCheck;
+    publicinboxwatch.watchspam = cfg.watch.watchSpam;
+  } cfg.config;
+
+  configList = configToList configFull;
+
+  gitConfig = key: val: ''
+    ${pkgs.git}/bin/git config --add --file $out ${escapeShellArgs [ key val ]}
+  '';
+
+  configFile = pkgs.runCommand "public-inbox-config" {}
+    (concatStrings (map ({ name, value }: gitConfig name value) configList));
+
+  environment = {
+    PI_EMERGENCY = "/var/lib/public-inbox/emergency";
+    PI_CONFIG = configFile;
+  };
+
+  envList = mapAttrsToList (n: v: "${n}=${v}") environment;
+
+  # Can't use pkgs.linkFarm,
+  # because Postfix rejects .forward if it's a symlink.
+  home = pkgs.runCommand "public-inbox-home" {
+    forward = ''
+      |"env ${concatStringsSep " " envList} PATH=\"${makeBinPath cfg.path}:$PATH\" ${cfg.package}/bin/public-inbox-mda
+    '';
+    passAsFile = [ "forward" ];
+  } ''
+    mkdir $out
+    ln -s /var/lib/public-inbox/spamassassin $out/.spamassassin
+    cp $forwardPath $out/.forward
+  '';
+
+  psgi = pkgs.substituteAll {
+    src = ./public-inbox.psgi;
+    perl = "${cfg.package.fullperl}/bin/perl";
+    path = cfg.http.mount;
+  };
+
+  descriptionFile = { description, ... }:
+    pkgs.writeText "description" description;
+
+  enableWatch = (any (i: i.watch != []) (attrValues cfg.inboxes))
+                || (cfg.watch.watchSpam != null);
+
+  useSpamAssassin = cfg.mda.spamCheck == "spamc" ||
+                    cfg.watch.spamCheck == "spamc";
+
+in
+
+{
+  options = {
+    services.public-inbox = {
+      enable = mkEnableOption "the public-inbox mail archiver";
```

Comment:
Can you add some more info, perhaps upstream URL? The name "public-inbox" is not search friendly, I don't know what this is 😊

Reply:
It isn’t? When I do a web search for public-inbox (using a non-bubbling search engine) the project’s README is literally the first result. https://public-inbox.org/README.html

Reply:
In my duck-duck-go search it is in 4th place. Google place it in 1st, and with a better title and text snippet. I think at least for duck-duck-go users it would help to make it obvious what this module is talking about.

Reply:
I have never seen a URL in a `mkEnableOption`, and just looked through 30 or so of them and didn’t see any there either. So I don’t think it makes sense to add one for this one module when we don’t do so elsewhere.

In Nixpkgs, package URLs go in the `meta` attribute for the package, and `pkgs.public-inbox.meta.homepage` is indeed set to the correct value.